src/pages/Account/Index.tsx

```diff
@@ -48,19 +49,40 @@ export function accountPagePath(isObject: boolean) {
 }
 
 export default function AccountPage({isObject = false}: AccountPageProps) {
+  const navigate = useNavigate();
   const isGraphqlClientSupported = useGetIsGraphqlClientSupported();
   const maybeAddress = useParams().address;
   const address =
     maybeAddress !== undefined
       ? AccountAddress.from(maybeAddress).toStringLong()
       : "";
-  let loadingFunction;
-  if (isObject) {
-    loadingFunction = useGetAccountResources;
-  } else {
-    loadingFunction = useGetAccount;
-  }
-  const {data, error, isLoading} = loadingFunction(address);
+  const {
+    data: objectData,
+    error: objectError,
+    isLoading: objectIsLoading,
+  } = useGetAccountResources(address, {retry: false});
+  const {
+    data: accountData,
+    error: accountError,
+    isLoading: accountIsLoading,
+  } = useGetAccount(address, {retry: false});
+
+  const isLoading = objectIsLoading || accountIsLoading;
+  const data = isObject ? objectData : accountData;
+  const error = isObject ? objectError : accountError;
+
+  useEffect(() => {
+    // If we are on the account page, we might be loading an object. This
+    // handler will redirect to the object page if no account exists but an
+    // object does.
+    if (isObject) return;
```

Comment:
`return` in `useEffect` is used for cleanup functions. It might work without errors, but not guaranteed in future updates. Better to do something like:
```tsx
if (!isObject && !isLoading) { ... }
```